resources/resource_factories/create/create.go

```diff
@@ -126,3 +144,136 @@ func (c *Client) FromString(targetPath, content string) (resource.Resource, erro
 			})
 	})
 }
+
+// FromRemote expects one or n-parts of a URL to a resource
+// If you provide multiple parts they will be joined together to the final URL.
+func (c *Client) FromRemote(args ...interface{}) (resource.Resource, error) {
+	uri, headers := toURLAndHeaders(args)
+	rURL, err := url.Parse(uri)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse URL for resource %s", uri)
+	}
+
+	resourceID := helpers.MD5String(uri)
```

Comment:
We need to somehow include the header info in this cache key. It's probably simpler to use `helpers.HashString(args...)`.